plugins/wporg-5ftf/includes/pledge.php

```diff
@@ -117,19 +125,158 @@ function register_custom_post_type() {
  * @return void
  */
 function register_custom_post_status() {
+	$label_count = _n_noop(
+		'Deactivated <span class="count">(%s)</span>',
+		'Deactivated <span class="count">(%s)</span>',
+		'wporg-5ftf'
+	);
 	register_post_status(
-		FiveForTheFuture\PREFIX . '-deactivated',
+		DEACTIVE_STATUS,
 		array(
-			'label'       => __( 'Deactivated', 'wporg-5ftf' ),
-			'label_count' => _n_noop( 'Deactivated <span class="count">(%s)</span>', 'Deactivated <span class="count">(%s)</span>', 'wporg-5ftf' ),
-			'public'      => false,
-			'internal'    => false,
-			'protected'   => true,
-			CPT_ID        => true, // Custom parameter to streamline its use with the Pledge CPT.
+			'label'                  => __( 'Deactivated', 'wporg-5ftf' ),
+			'label_count'            => $label_count,
+			'public'                 => false,
+			'internal'               => false,
+			'protected'              => true,
+			'show_in_admin_all_list' => false,
 		)
 	);
 }
 
+/**
+ * Inject deactivate/reactivate actions into row actions.
+ *
+ * @return array An array of row action links.
+ */
+function add_row_action( $actions, $post ) {
+	// Not a pledge, or can't edit the post.
+	if ( CPT_ID !== $post->post_type || ! current_user_can( 'edit_post', $post->ID ) ) {
+		return $actions;
+	}
+	$post_type_object = get_post_type_object( $post->post_type );
+	if ( DEACTIVE_STATUS === $post->post_status ) {
+		$actions['reactivate'] = sprintf(
+			'<a href="%s" style="color:#297531;" aria-label="%s">%s</a>',
+			wp_nonce_url( admin_url( sprintf( $post_type_object->_edit_link . '&amp;action=reactivate', $post->ID ) ), 'reactivate-post_' . $post->ID ),
+			/* translators: %s: Post title. */
+			esc_attr( sprintf( __( 'Reactivate pledge &#8220;%s&#8221;', 'wporg-5ftf' ), $post->post_title ) ),
+			__( 'Reactivate', 'wporg-5ftf' )
+		);
+	} else {
+		unset( $actions['trash'] );
```

Comment:
Admins can still directly trash pledges from the regular edit screen or via bulk actions, but for the sake of having fewer actions here I've removed it.